### ISSUE TITLE
[Android] Updates Arriving Out-of-Order

### ISSUE THREAD

**Issue by rarcher** [dotnet-bluetooth-le/dotnet-bluetooth-le#546]
In a case where a Bluetooth device triggers multiple updates on a single characteristic in quick succession, the resulting callbacks tend to arrive out-of-order on Android.

Digging into the relevant code and documentation, I believe this is the cause:
```
var connectGatt = BluetoothDevice.ConnectGatt(Application.Context, connectParameters.AutoConnect, _gattCallback);
```
```
Handler: The handler to use for the callback. If null, callbacks will happen on  an un-specified background thread.
```

The plugin's call to the native `ConnectGatt` function does not supply a `handler` parameter, and according to the Android docs, this causes the callback to happen on "an un-specified background thread".  Testing has shown that (on some devices, at least) there are multiple threads that can potentially handle the callback.  This results in a data race, if the Bluetooth device is sending multiple updates on the same characteristic.

TLDR; Please provide a way to specify a `handler` parameter when `ConnectGatt` is called, or alternatively supply an arbitrarily-chosen one automatically.
